javascript/auth.js

```diff
@@ -1,26 +1,149 @@
+    isPasswordConfirmError === false
+  ) {
+    submitButton.disabled = false;
+    submitButton.style.backgroundColor = "var(--brand-blue)";
+  } else {
+    submitButton.disabled = true;
+    submitButton.style.backgroundColor = "var(--gray-400)";
+  }
+};
+
+const handleEmailValidation = () => {
+  const emailValue = emailInput.value;
+  const regex =
+    /^[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*\.[a-zA-Z]{2,3}$/i;
+  hideError(emailInput, "email-error");
+  hideError(emailInput, "email-empty");
+  if (!emailValue) {
+    showError(emailInput, "email-empty");
+    isEmailError = true;
+  } else {
+    hideError(emailInput, "email-empty");
+    if (regex.test(emailValue)) {
+      hideError(emailInput, "email-error");
+      isEmailError = false;
+    } else {
+      showError(emailInput, "email-error");
+      isEmailError = true;
+    }
+  }
+
+  nameInput ? signupActivateSubmit() : loginActivateSubmit();
+};
+
+const handleNameValidation = () => {
+  const nameValue = nameInput.value;
+
+  if (!nameValue) {
+    showError(nameInput, "name-empty");
+    isNameError = true;
+  } else {
+    hideError(nameInput, "name-empty");
+    isNameError = false;
+  }
+  signupActivateSubmit();
+};
+
+const handlePasswordValidation = () => {
+  const passwordValue = passwordInput.value;
+
+  hideError(passwordInput, "password-empty");
+  hideError(passwordInput, "password-error");
+
+  if (!passwordValue) {
+    showError(passwordInput, "password-empty");
+    isPasswordError = true;
+  } else {
+    hideError(passwordInput, "password-empty");
+    if (passwordValue.length >= 8) {
+      hideError(passwordInput, "password-error");
+      isPasswordError = false;
+    } else {
+      showError(passwordInput, "password-error");
+      isPasswordError = true;
+    }
+  }
+  nameInput ? signupActivateSubmit() : loginActivateSubmit();
+};
+
+const handlePasswordConfirmValidation = () => {
+  const passwordConfirmValue = passwordConfirmInput.value;
+  const passwordValue = passwordInput.value;
+  const isMatch = passwordConfirmValue === passwordValue;
+
+  if (!passwordValue || !isMatch) {
+    showError(
+      passwordConfirmInput,
+      "password-confirm-error",
+      isPasswordConfirmError
+    );
+    isPasswordConfirmError = true;
+  } else {
+    hideError(
+      passwordConfirmInput,
+      "password-confirm-error",
+      isPasswordConfirmError
+    );
+    isPasswordConfirmError = false;
+  }
+  signupActivateSubmit();
+};
+
+const handleSubmit = (e) => {
+  e.preventDefault();
+  if (nameInput) {
+    location.href = "/auth/login.html";
+  } else {
+    location.href = "/item.html";
   }
-  toggleImg.src = "/image/btn_visibility_off.png";
-  toggleImg.alt = "invisible";
-  authPasswordInput.type = "password";
 };
 
 passwordToggleButton.forEach((passwordBtn) => {
   passwordBtn.addEventListener("click", handlePasswordToggleButton);
 });
+
+emailInput.addEventListener("focusout", handleEmailValidation);
```

Comment:
연관있는 로직끼리는 가까운 라인에 위치시켜주시면 가독성이 좋아집니다 ㅎㅎ 

```suggestion
const handleEmailValidation = () => {
  const emailValue = emailInput.value;
  const regex =
    /^[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*\.[a-zA-Z]{2,3}$/i;
  hideError(emailInput, "email-error");
  hideError(emailInput, "email-empty");
  if (!emailValue) {
    showError(emailInput, "email-empty");
    isEmailError = true;
  } else {
    hideError(emailInput, "email-empty");
    if (regex.test(emailValue)) {
      hideError(emailInput, "email-error");
      isEmailError = false;
    } else {
      showError(emailInput, "email-error");
      isEmailError = true;
    }
  }

  nameInput ? signupActivateSubmit() : loginActivateSubmit();
};
emailInput.addEventListener("focusout", handleEmailValidation);
```